### ISSUE TITLE
M7: Lasso Regression with TimeSeriesSplit (Multi-Variable Insights)

### ISSUE THREAD

**Issue by Sturmi77** [Sturmi77/correlcore#144]
## Context

M3 computes univariate correlations (one tag ↔ one metric). M8 upgrades to **multivariate Lasso regression** over all variables simultaneously, separating confounded effects (e.g., sport and outdoor walks often co-occur) and producing a ranked list of independent predictors.

Ref: DESIGN_DOCUMENT.md §6 M8 — *"Multiple Regression (Lasso) over all variables"*

---

### Feature Matrix Construction

```python
import pandas as pd
from sklearn.linear_model import LassoCV
from sklearn.model_selection import TimeSeriesSplit
from sklearn.preprocessing import StandardScaler

def build_feature_matrix(df: pd.DataFrame) -> tuple[pd.DataFrame, pd.Series]:
    """Build X (features) and y (target) for Lasso regression."""
    tag_cols = [c for c in df.columns if c.startswith('tag_')]
    feature_cols = tag_cols + ['energy', 'stress', 'sleep_minutes', 'sleep_quality']
    # work_context as one-hot encoded dummies
    work_dummies = pd.get_dummies(df['work_context'], prefix='ctx', drop_first=True)
    X = pd.concat([df[feature_cols], work_dummies], axis=1).fillna(0)
    y = df['mood_score']
    return X, y
```

### Lasso with TimeSeriesSplit (mandatory — see ADR-0016)

```python
# MUST use TimeSeriesSplit — standard KFold leaks future data into training
tscv = TimeSeriesSplit(n_splits=5)
scaler = StandardScaler()
X_scaled = scaler.fit_transform(X)

model = LassoCV(
    cv=tscv,
    max_iter=10_000,
    random_state=42,
)
model.fit(X_scaled, y)

# Coefficients: non-zero = selected by Lasso
coefficients = pd.Series(model.coef_, index=X.columns)
selected = coefficients[coefficients != 0].sort_values(key=abs, ascending=False)
```

### Storing M8 Insights

Lasso insights are stored as a new `insight_type = 'lasso'` alongside existing M3 `insight_type = 'pointbiserial'` / `'spearman'`. The `insights` table must be extended:

```sql
-- New migration: add insight_type column
ALTER TABLE insights ADD COLUMN insight_type VARCHAR(32) NOT NULL DEFAULT 'pointbiserial';
ALTER TABLE insights ADD COLUMN model_version VARCHAR(16) DEFAULT NULL; -- e.g. 'lasso_v1'
```

---

## Acceptance Criteria

- [ ] `LassoCV` with `TimeSeriesSplit(n_splits=5)` implemented in `backend/workers/analytics_m8.py`
- [ ] Features include: all tags (binary), energy, stress, sleep_minutes, sleep_quality, work_context (one-hot)
- [ ] `StandardScaler` applied before Lasso (required for L1 regularisation fairness)
- [ ] Results stored in `insights` table with `insight_type = 'lasso'`
- [ ] New Alembic migration adds `insight_type` and `model_version` columns with backward-compatible defaults
- [ ] Worker skips user if fewer than 90 entries (insufficient for 5-fold TimeSeriesSplit)
- [ ] Lasso results are reproducible: `random_state=42`, same input → identical coefficients
- [ ] Unit test: synthetic dataset where `tag_sport` has a known effect — assert `coef['tag_sport'] > 0`
- [ ] `scikit-learn` added to `pyproject.toml` (if not already present)
- [ ] `ruff check` / `mypy --strict` / `pytest` green
- [ ] CHANGELOG entry

## Minimum Data Requirement

`n ≥ 90 entries` for M8 features (5-fold TimeSeriesSplit requires at least ~18 entries per fold; 90 provides stable cross-validation).

## Out of Scope

- Lag features (separate issue)
- LLM statement generation (separate issue)
- Frontend visualisation of Lasso coefficients (extends Issue #19)

## References

- DESIGN_DOCUMENT.md §6 M8
- ADR-0016 `docs/adr/0016-timeseries-split-ml-models.md` (see this PR)
- Issue #16 (M3 univariate correlations — runs in parallel, not replaced)

**Comment by Sturmi77** [comment]
Closed via M7 Sprint 5: Lasso + TimeSeriesSplit shipped on main (PR #223). Insight type symptom_cluster with payload.method=lasso. See docs/M7_SPRINT_STATUS.md and docs/quality/M7_QUALITY_GATE.md.
